docs/zq/language.md

```diff
@@ -1,341 +1,1970 @@
+```
+over ... with <elem> [, <elem> ...] => ( <query> )
+```
+where `<elem>` has either an assignment form
+```
+<var>=<expr>
+```
+or a field reference form
+```
+<field>
+```
+For each input value to the outer scope, the assignment form create a binding
+between each `<expr>` evaluated in the outer scope and each `<var>`, which
+represents a new symbol in the inner scope of the `<query>`.
+In the field reference form, a single identifier `<id>` refers to a field
+in the parent scope and makes that field's value available in the lateral scope
+with the same name.
+
+The `<query>`, which may be any Zed query, is evaluated once per outer value
+on the sequence generated by the `over` expression.  In the lateral scope,
+the value `this` refers to the inner sequence generated from the `over` expressions.
+This query runs to completion for each inner sequence and yields the results to
+its output as each inner sequence completes.
+
+## 9. Shaping
+
+Data that originates from heterogenous sources often is combined into a
+unified store like a data warehouse but has inconsistent structure and is thus
+difficult to reason about or query.  To unify disparate data sources,
+data is often cleaned up to fit into a well-defined set of data models.
+
+In Zed, this cleansing process is called "shaping" the data and Zed leverages
+its rich, super-structured type system to perform core aspects of
+data transformation.
+In a data model with nesting and multiple scalar types (such as Zed or JSON),
+shaping includes converting type of leaf fields, adding or removing fields
+to "fit" a given shape, and reordering fields.
+
+While shaping remains an active area of development, the core functions in Zed
+that currently perform shaping are:
+
+* [cast](functions/cast.md) - coerce a value to a different type
+* [crop](functions/crop.md) - remove fields from a value that are missing in a specified type
+* [fill](functions/fill.md) - add null values for missing fields
+* [order](functions/order.md) - reorder record fields
+* [shape](functions/shape.md) - apply cast, fill, and order
+
+They all have the same signature, taking two parameters: the value to be
+transformed and a type value for the target type.
+
+> Another type of transformation that's needed for shaping is renaming fields,
+> which is supported by the [rename operator](operators/rename.md#operator).
+> Also, the [yield operator](operators/yield.md#operator)
+> is handy for simply emitting new, arbitrary record literals based on
+> input values and mixing in these shaping functions in an embedded record literal.
+> The [fuse aggregate function](aggregates/fuse.md) is useful for fusing
+> values into a common schema within a group-by schema though a type is returned
+> rather than values.
+
+In the examples below, we will use the following named type `connection`
+that is stored in a file [connection.zed](connection.zed)
+and is included in the example Zed queries with `-I` option of `zq`:
+```
```

Comment:
For example files like this, we should use the content here in the Markdown rather than duplicate it in a separate file.

To do that, change `mdtest-command dir=docs/zq` to just `mdtest-command` everywhere and then add `mdtest-input <filename>` to the fenced code blocks for the example files, like this:
~~~suggestion
```mdtest-input connection.zed
~~~

Reply:
cool!

Reply:
done